packages/medusa/src/services/totals.ts

```diff
@@ -140,23 +140,16 @@ class TotalsService extends BaseService {
     if (
       !cartOrOrder.region.automatic_taxes &&
       !forceTaxes &&
-      "payment_session" in cartOrOrder
+      cartOrOrder.object === "cart"
```

Comment:
tiny nit: we could follow the same approach here as you did with the `isCartCompletionStrategy` type guard function and add a `isCart()` and `isOrder()` methods

Reply:
I think that due to the way objects are being passed around and how things are sometimes being returned as `Partial<Cart>` we don't necessarily pass an object that would allow usage of `instanceof`.

We could definitely add something like this though:
```
export function isCart(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  object: any
): object is Cart {
  return typeof object.object === "cart"
}
```